fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java

```diff
@@ -0,0 +1,503 @@
+            String columnName = partHiveColumn.getName();
+            Column partColumn = this.nameToColumn.get(columnName);
+            if (partColumn == null) {
+                throw new DdlException("Partition column [" + columnName + "] must exist in column list");
+            } else {
+                this.partColumnNames.add(columnName);
+            }
+        }
+
+        for (FieldSchema s : unPartHiveColumns) {
+            this.dataColumnNames.add(s.getName());
+        }
+
+        Configuration conf = new Configuration();
+        HoodieTableMetaClient metaClient = HoodieTableMetaClient.builder().setConf(conf).setBasePath(hudiBasePath).build();
+        TableSchemaResolver schemaUtil = new TableSchemaResolver(metaClient);
+        Schema tableSchema;
+        try {
+            tableSchema = HoodieAvroUtils.createHoodieWriteSchema(schemaUtil.getTableAvroSchemaWithoutMetadataFields());
+        } catch (Exception e) {
+            throw new DdlException("Cannot get hudi table schema.");
+        }
+        for (Column column : this.fullSchema) {
+            if (!column.isAllowNull()) {
+                throw new DdlException(
+                        "Hudi extern table does not support no-nullable column: [" + column.getName() + "]");
+            }
+            Schema.Field hudiColumn = tableSchema.getField(column.getName());
+            if (hudiColumn == null) {
+                throw new DdlException("Column [" + column.getName() + "] not exists in hudi.");
+            }
+            Schema.Type hudiType = hudiColumn.schema().getTypes().get(1).getType();
+            Set<PrimitiveType> validColumnTypes = getValidColumnType(hudiType);
+            if (!validColumnTypes.contains(column.getPrimitiveType())) {
+                throw new DdlException("Can not convert hudi column type [" + hudiType + "] to " +
+                        "starrocks type [" + column.getPrimitiveType() + "], column name: " + column.getName());
+            }
+        }
+
+        if (!copiedProps.isEmpty()) {
+            throw new DdlException("Unknown table properties: " + copiedProps.toString());
+        }
+    }
+
+    private Set<PrimitiveType> getValidColumnType(Schema.Type type) {
+        if (type == null) {
+            return Sets.newHashSet();
+        }
+
+        switch (type) {
+            case BOOLEAN:
+                return Sets.newHashSet(PrimitiveType.BOOLEAN);
+            case INT:
+                return Sets.newHashSet(PrimitiveType.INT, PrimitiveType.TINYINT, PrimitiveType.SMALLINT);
+            case LONG:
+                return Sets.newHashSet(PrimitiveType.BIGINT);
+            case FLOAT:
+                return Sets.newHashSet(PrimitiveType.FLOAT);
+            case DOUBLE:
+                return Sets.newHashSet(PrimitiveType.DOUBLE);
+            case STRING:
+                return Sets.newHashSet(PrimitiveType.VARCHAR, PrimitiveType.CHAR);
+            case FIXED:
+            case ARRAY:
```

Comment:
https://github.com/StarRocks/starrocks/blob/8913154214f33be88147dae51d4850a4d6c92121/fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java#L162

we have already support array

Reply:
done, array type added